.github/workflows/scripts/add-labels.sh

```diff
@@ -16,15 +16,59 @@
 #
 #
 
-if [ -z "${ISSUE}" ] || [ -z "${COMMENT}" ]; then
+set -euo pipefail
+
+if [[ -z "${ISSUE:-}" || -z "${COMMENT:-}" || -z "${SENDER:-}" ]]; then
+    echo "At least one of ISSUE, COMMENT, or SENDER has not been set, please ensure each is set."
     exit 0
 fi
 
-declare -A labels
-labels["/good-first-issue"]="good first issue"
-labels["/help-wanted"]="help wanted"
-
+CUR_DIRECTORY=$(dirname "$0")
 
-if [ -n "${labels["${COMMENT}"]}" ] ; then
-    gh issue edit "${ISSUE}" --add-label "${labels["${COMMENT}"]}"
+if [[ ${COMMENT:0:6} != "/label" ]]; then
+    echo "Comment is not a label comment, exiting."
+    exit 0
 fi
+
+declare -A COMMON_LABELS
+COMMON_LABELS["good-first-issue"]="good first issue"
+COMMON_LABELS["help-wanted"]="help wanted"
+
+LABELS=$(echo "${COMMENT}" | sed -E 's%^/label%%')
+
+for LABEL_REQ in ${LABELS}; do
+    LABEL=$(echo "${LABEL_REQ}" | sed -E s/^[+-]?//)
+    SHOULD_ADD=true
+
+    if [[ "${LABEL_REQ:0:1}" = "-" ]]; then
+        SHOULD_ADD=false
+    fi
+
+    if [[ -v COMMON_LABELS["${LABEL}"] ]]; then
+        if [[ ${SHOULD_ADD} = true ]]; then
+            gh issue edit "${ISSUE}" --add-label "${COMMON_LABELS["${LABEL}"]}"
+        else
+            gh issue edit "${ISSUE}" --remove-label "${COMMON_LABELS["${LABEL}"]}"
+        fi
+        continue
+    fi
+
+    OWNERS=$(COMPONENT="${LABEL}" bash "${CUR_DIRECTORY}/get-codeowners.sh")
+
+    if [[ -z "${OWNERS}" ]]; then
+        echo "\"${LABEL}\" is a component with no code owners or doesn't correspond to a component, skipping."
+        continue
+    fi
```

Comment:
Why do we need to skip setting label in this case? I think it's still useful to put labels even if no one is pinged. In that case we don't ned to copy paste code from ping-codeowners.sh to ping-codeowners.sh

Reply:
That's a good point. I've updated it accordingly.

I would still like `get-codeowners.sh` since it is used in other workflows where we ping code owners, but I've removed it from this PR. I'll update `ping-codeowners.sh` and `mark-issues-as-stale.sh` to use it in a follow-up PR that includes a few other small refactors.

Reply:
Sounds like a good plan!